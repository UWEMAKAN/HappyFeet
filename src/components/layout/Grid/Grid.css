.Grid {
  grid-column: 1 / 2;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  row-gap: 5rem;
  background-color: #fff;
}

@media screen and (max-width: 900px) {
  .Grid {
    grid-template-columns: repeat(2, 1fr);
    row-gap: 9rem;
  }
}

@media screen and (max-width: 800px) {
  .Grid {
    row-gap: 6rem;
  }
}

@media screen and (max-width: 700px) {
  .Grid {
    grid-template-columns: repeat(2, 1fr);
    row-gap: 4rem;
  }
}

@media screen and (max-width: 450px) {
  .Grid {
    border: none;
    grid-template-rows: min-content;
    row-gap: 2rem;
  }
}
