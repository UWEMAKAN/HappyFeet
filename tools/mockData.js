module.exports = {
  shoes: [
    {
      id: 1,
      title: 'Black Snickers',
      brand: 'Others',
      image: 'http://localhost:3001/assets/images/others-black-1-600x600.jpg',
      color: 'Black',
      price: 1295,
      discount: 35,
      rating: 4.3,
    },
    {
      id: 2,
      title: 'Black Snickers',
      brand: 'Others',
      image: 'http://localhost:3001/assets/images/others-black-2-600x600.jpg',
      color: 'Black',
      price: 1325,
      discount: 35,
      rating: 4.3,
    },
    {
      id: 3,
      title: 'White Snickers',
      brand: 'Nike',
      image: 'http://localhost:3001/assets/images/nike-white-1-600x600.jpg',
      color: 'White',
      price: 1450,
      discount: 40,
      rating: 4.6,
    },
    {
      id: 4,
      title: 'High Hills',
      brand: 'Others',
      image: 'http://localhost:3001/assets/images/others-black-hills-1-600x600.jpg',
      color: 'Black',
      price: 1375,
      discount: 40,
      rating: 4.3,
    },
    {
      id: 5,
      title: 'Black Leather Shoes',
      brand: 'Others',
      image: 'http://localhost:3001/assets/images/others-black-leather-1-600x600.jpg',
      color: 'Black',
      price: 2425,
      discount: 30,
      rating: 4.3,
    },
    {
      id: 6,
      title: 'Blue Snickers',
      brand: 'Others',
      image: 'http://localhost:3001/assets/images/others-blue-1-600x600.jpeg',
      color: 'Blue',
      price: 3420,
      discount: 35,
      rating: 4.3,
    },
  ]
};
