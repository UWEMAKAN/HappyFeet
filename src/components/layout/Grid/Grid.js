import React from 'react';
import propTypes from 'prop-types';
import Card from '../../Card/Card';
import NoResultsFound from '../../Common/NoResultsFound/NoResultsFound';
import './Grid.css';
import Spinner from '../../Common/Spinner/Spinner';

const Grid = ({ shoes, eventFired, clickHandler }) => {
  const shoesCards = shoes.map((shoe) => (
    <Card
      id={`${shoe.id}`}
      key={shoe.id}
      image={shoe.image}
      title={shoe.title}
      price={shoe.price}
      discount={shoe.discount}
      rating={shoe.rating}
      clickHandler={clickHandler}
    />
  ));

  let returnValue;
  if (eventFired) {
    returnValue = <Spinner />;
  } else {
    returnValue = shoes.length ? <div className="Grid">{shoesCards}</div> : <NoResultsFound message="No Results Found" />;
  }

  return (
    <>
      {returnValue}
    </>
  );
};

Grid.propTypes = {
  shoes: propTypes.instanceOf(Array).isRequired,
  eventFired: propTypes.bool.isRequired,
  clickHandler: propTypes.func.isRequired
};

export default Grid;
