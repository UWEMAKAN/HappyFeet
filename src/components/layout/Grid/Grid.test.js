import React from 'react';
import { shallow } from 'enzyme';
import Grid from './Grid';
import { shoes } from '../../../../tools/mockData';

const renderGrid = (args) => {
  const defaultProps = {
    shoes: [],
    eventFired: false,
    clickHandler: jest.fn()
  };
  const props = { ...defaultProps, ...args };
  // eslint-disable-next-line react/jsx-props-no-spreading
  return shallow(<Grid {...props} />);
};

describe('Grid Component', () => {
  test('should render Spinner when eventFired is true', () => {
    const eventFired = true;
    const wrapper = renderGrid({ shoes, eventFired });
    expect.assertions(2);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toEqual('<Spinner />');
  });

  test('should render NoResultsFound when eventFired is false and shoes.length is 0', () => {
    const eventFired = false;
    const wrapper = renderGrid({ shoes: [], eventFired });
    expect.assertions(2);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.text()).toEqual('<NoResultsFound />');
  });

  test('should render a list of Cards when eventFired is false and shoes.length is > 0', () => {
    const eventFired = false;
    const wrapper = renderGrid({ shoes, eventFired });
    // expect.assertions(2);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.getElements()[0].props.children.props.children.length).toBe(6);
  });
});
