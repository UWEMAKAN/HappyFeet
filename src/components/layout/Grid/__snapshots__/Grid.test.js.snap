// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Component should render NoResultsFound when eventFired is false and shoes.length is 0 1`] = `ShallowWrapper {}`;

exports[`Grid Component should render Spinner when eventFired is true 1`] = `ShallowWrapper {}`;

exports[`Grid Component should render a list of Cards when eventFired is false and shoes.length is > 0 1`] = `ShallowWrapper {}`;
